### PR TITLE
Fix HDA HDMI only topology load issue

### PR DESCRIPTION
**Pull request by xiulipan**
Fix https://github.com/thesofproject/sof/issues/1733

Also refine the hda-generic to sync with hda-generic-idisp to avoid future mis-sync.

### CODE REVIEW

**Comment by tlauda** [comment]
@xiulipan I think your changes break the topology.

File: tools/topology/sof-hda-generic-idisp.m4 [hunk 1]
```diff
@@ -71,42 +71,48 @@ dnl DAI_ADD(pipeline,
...
+#
 # PCM Low Latency, id 0
 dnl PCM_PLAYBACK_ADD(name, pcm_id, playback)
-PCM_PLAYBACK_ADD(HDMI1, 1, PIPELINE_PCM_2)
-PCM_PLAYBACK_ADD(HDMI2, 2, PIPELINE_PCM_3)
-PCM_PLAYBACK_ADD(HDMI3, 3, PIPELINE_PCM_4)
+
+PCM_PLAYBACK_ADD(HDMI1, 3, PIPELINE_PCM_7)
+PCM_PLAYBACK_ADD(HDMI2, 4, PIPELINE_PCM_8)
+PCM_PLAYBACK_ADD(HDMI3, 5, PIPELINE_PCM_9)
```
Comment (plbossart) [inline tools/topology/sof-hda-generic-idisp.m4]: no, this is not right. it makes no sense to 'align' generic and generic-disp on the pipeline indices. There are different for the simple reason that one has HDaudio codec capture/playback and the other doesn't.

File: tools/topology/sof-hda-generic-idisp.m4 [hunk 2]
```diff
@@ -71,42 +71,48 @@ dnl DAI_ADD(pipeline,
...
-DAI_CONFIG(HDA, 2, 3, iDisp3)
+
+# 3 HDMI/DP outputs (ID: 3,4,5)
+DAI_CONFIG(HDA, 4, 1, iDisp1)
+DAI_CONFIG(HDA, 5, 2, iDisp2)
+DAI_CONFIG(HDA, 6, 3, iDisp3)
+
+VIRTUAL_DAPM_ROUTE_OUT(iDisp1_out, HDA, 4, OUT, 7)
+VIRTUAL_DAPM_ROUTE_OUT(iDisp2_out, HDA, 5, OUT, 8)
+VIRTUAL_DAPM_ROUTE_OUT(iDisp3_out, HDA, 6, OUT, 9)
```
Comment (plbossart) [inline tools/topology/sof-hda-generic-idisp.m4]: this is going too far in the alignment. There's a reason why we kept the two files separate.

File: tools/topology/sof-hda-generic.m4 [hunk 3]
```diff
@@ -32,6 +32,11 @@ include(`platform/intel/intel-generic-dmic.m4')
 #
 # PCM0  <---> volume (pipe 1,2) <----> HDA Analog (HDA Analog playback/capture)
 # PCM1  <---> volume (pipe 3,4) <----> HDA Digital (HDA Digital playback/capture)
+
+include(`sof-hda-generic-idisp.m4')
```
Comment (plbossart) [inline tools/topology/sof-hda-generic.m4]: I don't think this makes sense either

Comment (xiulipan) [reply tools/topology/sof-hda-generic.m4]: This trick is common used in a few topology files
https://github.com/thesofproject/sof/blob/66da4e89a9c2fce8fe04624a1001f7b8111a40c2/tools/topology/sof-cml-rt5682-max98357a.m4#L7-L9
https://github.com/thesofproject/sof/blob/66da4e89a9c2fce8fe04624a1001f7b8111a40c2/tools/topology/sof-cml-demux-rt5682-max98357a.m4#L7-L9
It is simple to keep the same part in different topology is the same.

File: tools/topology/sof-hda-generic-idisp.m4 (hunk 2 above)
Comment (xiulipan) [reply tools/topology/sof-hda-generic-idisp.m4]: @plbossart 
I checked the original  sof-hda-generic-idisp.m4 code. The IDs and even the pipeline fmt is not in right order. I think no one have test this topology once.

**Comment by xiulipan** [comment]
@tlauda 
It seems ```
include(`sof-hda-generic-idisp.m4')``` will also include one dmic include. So I need to delete one.

File: tools/topology/sof-hda-generic-idisp.m4 (hunk 2 above)
Comment (xiulipan) [reply tools/topology/sof-hda-generic-idisp.m4]: @plbossart 
Re-check the machine driver implement. We change the dai id for DMIC in run time fix. I think this it the reason you are talking about.

**Comment by xiulipan** [comment]
@plbossart @tlauda 
Close as find some mismatch with machine driver in this solution.

**Pull request closed without merging**